client/src/components/TeaTypes/TeaTypeEditor/TeaTypeEditorContainer.jsx

```diff
@@ -149,17 +151,32 @@ export class TeaTypeEditorContainer extends React.Component {
     }
   };
 
+  serverErrorStateUpdate = () => {
+    this.setState(state => ({
+      inputValidation: {
+        ...state.inputValidation,
+        noDuplicate: false
+      },
+      flash: { name: "" }
+    }));
+  };
+
+  loadingStatusStateUpdate = () => {
+    this.setState({ loading: false });
+  };
+
   componentDidMount() {
+    this.props.getLoadingStatus;
+
     this.props.getTeaTypes(this.props.userID);
-    if (this.props.serverErrors && this.props.serverErrors.noDuplicate) {
-      this.setState(state => ({
-        inputValidation: {
-          ...state.inputValidation,
-          noDuplicate: false
-        },
-        flash: { name: "" }
-      }));
-    }
+
+    this.props.serverErrors &&
+      this.props.serverErrors.noDuplicate &&
+      this.serverErrorStateUpdate();
+
+    this.props.loading &&
+      this.props.loading === "complete" &&
+      this.loadingStatusStateUpdate();
```

Comment:
This looks to be the only place in any component where you're actually checking on loading status. This would suggest to me that it should not be in the Redux state in the first place, as the loading status is only relevant to this component.

Instead of polling for loading status upon every component update, I would simply ensure `getTeaTypes` returns a promise (`API.get()` returns a promise) add a `.then()` to the end of the call on line 171, and unset a _local_ `loading` state value instead.